### PR TITLE
feat: add multi-file introspection to schema engine cli

### PR DESCRIPTION
**Pull request by Weakky**
## Overview

Closes https://github.com/prisma/team-orm/issues/1151

### CODE REVIEW

File: schema-engine/cli/tests/cli_tests.rs [hunk 1]
```diff
@@ -659,7 +659,7 @@ fn introspect_postgres(api: TestApi) {
...
+            {"jsonrpc":"2.0","result":{"datamodels":[{"content":"generator js {\n  provider        = \"prisma-client-js\"\n  previewFeatures = [\"views\"]\n}\n\ndatasource db {\n  provider = \"postgres\"\n  url      = env(\"TEST_DATABASE_URL\")\n}\n\nmodel A {\n  id   Int     @id @default(autoincrement())\n  data String?\n}\n\n/// The underlying view does not contain a valid unique identifier and can therefore currently not be handled by Prisma Client.\nview B {\n  col Int?\n\n  @@ignore\n}\n","path":"schema.prisma"}],"views":[{"definition":"SELECT\n  1 AS col;","name":"B","schema":"public"}],"warnings":"*** WARNING ***\n\nThe following views were ignored as they do not have a valid unique identifier or id. This is currently not supported by Prisma Client. Please refer to the documentation on defining unique identifiers in views: https://pris.ly/d/view-identifiers\n  - \"B\"\n"},"id":1}
```
Comment (SevInf) [inline schema-engine/cli/tests/cli_tests.rs:662]: Can we rename to `schemaFiles` or `schamas` while we are it?

**Review by SevInf — COMMENTED** [review]
Nit on the naming and one more thing: can we have a little bit more tests for re-introspection of existing models and new models being added?

File: schema-engine/cli/tests/cli_tests.rs (hunk 1 above)
Comment (Weakky) [reply schema-engine/cli/tests/cli_tests.rs:662]: Renamed it to `{ schema: { files: [] } }` for symmetry with the input

**Review by SevInf — APPROVED** [review]
LGTM after integration PR for client lined up

**Comment by Weakky** [comment]
Follow-up PR: https://github.com/prisma/prisma-engines/pull/4892

**Pull request closed without merging**